### PR TITLE
fix: no timeout for urlopen issue #526

### PR DESCRIPTION
**Pull request by grzracz**

### CODE REVIEW

File: algosdk/kmd.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ def kmd_request(self, method, requrl, params=None, data=None):
             requrl (str): url for the request
             params (dict, optional): parameters for the request
             data (dict, optional): data in the body of the request
+            timeout (int, optional): request timeout in seconds
```
Comment (jannotti) [inline algosdk/kmd.py:38]: Let's document the 30 second default here.

File: algosdk/v2client/algod.py [hunk 2]
```diff
@@ -72,6 +73,7 @@ def algod_request(
             data (bytes, optional): data in the body of the request
             headers (dict, optional): additional header for request
             response_format (str, optional): format of the response
+            timeout (int, optional): request timeout in seconds
```
Comment (jannotti) [inline algosdk/v2client/algod.py:76]: And here.

**Review by jannotti — COMMENTED** [review]
`urlopen` docs say: The optional timeout parameter specifies a timeout in seconds for blocking operations like the connection attempt (if not specified, the global default timeout setting will be used). Any idea what that global is?

Regardless, it seems right to me to let the caller have access to the setting. I'd like @jasonpaulos to weigh in.

Another possible is to allow arbitrary `**kwargs`, and pass them through, if there are other useful features of `urlopen` that it would give access to. It doesn't seem like it though, the rest appears to about SSL details.

**Comment by grzracz** [comment]
Seems like the default timeout is implementation-dependent and in Python it just never times out:
https://stackoverflow.com/questions/29649173/what-is-the-global-default-timeout

File: algosdk/kmd.py (hunk 1 above)
Comment (grzracz) [reply algosdk/kmd.py:38]: Would be inconsistent since "response_format" is default "json" and is not documented in the docstring either

Comment (jannotti) [reply algosdk/kmd.py:38]: Agreed, the signature shows in the docs, so the defaults are visible.

**Pull request merged**